### PR TITLE
Port yuzu-emu/yuzu#4290 and yuzu-emu/yuzu#4318: Changes for C++20 support

### PR DESCRIPTION
**Pull request by FearlessTobi**
See https://github.com/yuzu-emu/yuzu/pull/4290 and https://github.com/yuzu-emu/yuzu/pull/4318 for more details.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/5459)

### CODE REVIEW

**Comment by lioncash** [comment]
You'll need to increase the minimum CMake version required to at least 3.12.4 at the top of the main CMakeLists file. IMO it can probably be incremented to 3.15.0, so the conditional at the top can be applied universally.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
-# CMake 3.8 required for 17 to be a valid value for CXX_STANDARD
-cmake_minimum_required(VERSION 3.8)
+# CMake 3.12.4 required for 20 to be a valid value for CXX_STANDARD
+cmake_minimum_required(VERSION 3.15)
 if (${CMAKE_VERSION} VERSION_GREATER_EQUAL 3.15)
```
Comment (lioncash) [inline CMakeLists.txt:3]: This if statement can be removed, and the contents of it can be made unconditional now, since 3.15 is now a requirement.

**Comment by FearlessTobi** [comment]
CI passes now.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,9 +1,9 @@
-# CMake 3.8 required for 17 to be a valid value for CXX_STANDARD
-cmake_minimum_required(VERSION 3.8)
-if (${CMAKE_VERSION} VERSION_GREATER_EQUAL 3.15)
-    # Don't override the warning flags in MSVC:
-    cmake_policy(SET CMP0092 NEW)
-endif ()
+# CMake 3.12.4 required for 20 to be a valid value for CXX_STANDARD
+cmake_minimum_required(VERSION 3.15)
+# Add MSVC runtime library selection flags automatically:
+cmake_policy(SET CMP0091 OLD)
```
[the remark this answers is not in the document]
Comment (FearlessTobi) [reply CMakeLists.txt:4]: Not using this policy breaks cryptopp building.

File: src/citra_qt/game_list.cpp [hunk 3]
```diff
@@ -567,8 +567,8 @@ void GameList::AddCustomDirPopup(QMenu& context_menu, QModelIndex selected) {
 void GameList::AddPermDirPopup(QMenu& context_menu, QModelIndex selected) {
     const int game_dir_index = selected.data(GameListDir::GameDirRole).toInt();
 
-    QAction* move_up = context_menu.addAction(tr(u8"\U000025b2 Move Up"));
-    QAction* move_down = context_menu.addAction(tr(u8"\U000025bc Move Down "));
+    QAction* move_up = context_menu.addAction(tr("\u25B2 Move Up"));
```
Comment (zhaowenlan1779) [inline src/citra_qt/game_list.cpp:570]:
```suggestion
    QAction* move_up = context_menu.addAction(tr("\u25b2 Move Up"));
```

**Review by zhaowenlan1779 — APPROVED** [review]

**Comment by zhaowenlan1779** [comment]
Please resolve the conflicts

**Comment by FearlessTobi** [comment]
Will merge in 24 hours if no more reviews.

File: CMakeLists.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (vitor-k) [reply CMakeLists.txt:4]: cryptopp was updated since. Is this still the case?

File: CMakeLists.txt [hunk 4]
```diff
@@ -158,8 +158,17 @@ message(STATUS "Target architecture: ${ARCHITECTURE}")
 # Configure C++ standard
 # ===========================
 
-set(CMAKE_CXX_STANDARD 20)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
+# boost asio's concept usage doesn't play nicely with some compilers yet.
+add_definitions(-DBOOST_ASIO_DISABLE_CONCEPTS)
+if (MSVC)
+    add_compile_options($<$<COMPILE_LANGUAGE:CXX>:/std:c++latest>)
```
Comment (vitor-k) [inline CMakeLists.txt:164]: Is this still necessary? It seems that for MSVC 2022 or 2019 after 16.11 it's equivalent to /std:c++20 (with the exception of few features like std::format), so maybe just setting the cmake c++ standard is enough?

Comment (FearlessTobi) [reply CMakeLists.txt]: Let's try

**Review by vitor-k — APPROVED** [review]
The changes lgtm, but I'm not sure if the CMP0091 change is necessary.

**Pull request merged**